app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepository.java

```diff
@@ -0,0 +1,365 @@
+                .setId(ebeanAlert.getUuid())
+                .setName(ebeanAlert.getName())
+                .setContext(ebeanAlert.getContext())
+                .setCluster(ebeanAlert.getCluster())
+                .setService(ebeanAlert.getService())
+                .setMetric(ebeanAlert.getMetric())
+                .setStatistic(ebeanAlert.getStatistic())
+                .setPeriod(Period.seconds(ebeanAlert.getPeriod()))
+                .setOperator(ebeanAlert.getOperator())
+                .setValue(new DefaultQuantity.Builder()
+                        .setValue(ebeanAlert.getQuantityValue())
+                        .setUnit(ebeanAlert.getQuantityUnit())
+                        .build())
+                .setExtensions(mergeExtensions(ebeanAlert.getNagiosExtension()))
+                .build();
+        return Optional.of(result);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AlertQuery createQuery() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Preparing query").log();
+        return new DefaultAlertQuery(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public QueryResult<Alert> query(final AlertQuery query) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Querying")
+                .addData("query", query)
+                .log();
+
+        // Create the base query
+        final PagedList<models.ebean.Alert> pagedAlerts = _alertQueryGenerator.createAlertQuery(query);
+
+        // Compute the etag
+        final SqlQuery sqlQuery = Ebean.createSqlQuery("SELECT CURRVAL('portal.alerts_etag_seq') AS etag;");
+        final Long etag = sqlQuery.findUnique().getLong("etag");
+
+        final List<Alert> values = new ArrayList<>();
+        pagedAlerts.getList().forEach(ebeanAlert -> {
+            final Alert result = new DefaultAlert.Builder()
+                    .setCluster(ebeanAlert.getCluster())
+                    .setContext(ebeanAlert.getContext())
+                    .setId(ebeanAlert.getUuid())
+                    .setMetric(ebeanAlert.getMetric())
+                    .setName(ebeanAlert.getName())
+                    .setOperator(ebeanAlert.getOperator())
+                    .setPeriod(Period.seconds(ebeanAlert.getPeriod()))
+                    .setService(ebeanAlert.getService())
+                    .setStatistic(ebeanAlert.getStatistic())
+                    .setValue(new DefaultQuantity.Builder()
+                            .setValue(ebeanAlert.getQuantityValue())
+                            .setUnit(ebeanAlert.getQuantityUnit())
+                            .build())
+                    .setExtensions(mergeExtensions(ebeanAlert.getNagiosExtension()))
+                    .build();
+            values.add(result);
+        });
+
+        // Transform the results
+        return new DefaultQueryResult<>(values, pagedAlerts.getTotalRowCount(), etag.toString());
```

Comment:
Can you file an issue against the project that we should obfuscate the etag value instead of returning the table version in plain? Please add a TODO here in the standard style and reference the issue (e.g. TODO(deepika): <desc> [ISSUE-?])